qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
This maintains that same issue, as `description` is being stored as a `NOVALUE` in the signature spec if it doesn't exist. Which raises a `TypeError: NOVALUE is not JSON serializable` on caching if you just add a `description` attribute to the action signature state. So it'd require something like:

```python
for group in 'inputs', 'outputs':
    for name, spec in getattr(action.signature, group).items():
        data = {
            'name': name,
            'repr': repr(spec.qiime_type),
        }
        if isinstance(spec.description, str):
            data['description'] = spec.description

        state['signature'][group].append(data)
```

Because we don't have access to the `ParameterSpec.NOVALUE` to compare it to, so we would have to see if it is a string every time. Unless that's fine? Having this helper that gathers all the descriptions seemed cleaner allowing, as it allows `description = descriptions.get(name, q2cli.handlers.NoDefault)` to be used on handler generation.